### PR TITLE
:bug: Align region resolution in create/delete cloudformation stack commands

### LINKED ISSUE #2392: clusterawsadm delete-cloudformation-stack should not return error on successful deletion
/kind feature

When running `clusterawsadm bootstrap iam create-cloudformation-stack` clusterawsadm seems to use sensible defaults to create this cloud formation stack and succeeds:

```
 clusterawsadm bootstrap iam create-cloudformation-stack
Attempting to create AWS CloudFormation stack cluster-api-provider-aws-sigs-k8s-io


Following resources are in the stack: 

Resource                  |Type                                                                                |Status
AWS::IAM::InstanceProfile |control-plane.cluster-api-provider-aws.sigs.k8s.io                                  |CREATE_COMPLETE
(truncated)
```

However, the same sensible defaults do not delete the stack:
```
clusterawsadm bootstrap iam delete-cloudformation-stack
Error: no AWS region configured. Use --region or set AWS_REGION or DEFAULT_AWS_REGION environment variable
```

Passing the `--region` argument allows this command to succeed (it does delete the stack) but it also shows an error message:

```
clusterawsadm bootstrap iam delete-cloudformation-stack --region eu-west-1
Attempting to delete AWS CloudFormation stack cluster-api-provider-aws-sigs-k8s-io
Error: unable to describe stack resources: ValidationError: Stack with id cluster-api-provider-aws-sigs-k8s-io does not exist
	status code: 400, request id: 3f3cf351-480b-405c-b3d8-17ac183fb7fb

```

**Describe the solution you'd like**
clusterawsadmin should be able to create and delete the cloudformation stack using the same arguments consistently, and not show an error message after the stack is deleted.

**Anything else you would like to add:**

**Environment:**

- Cluster-api-provider-aws version: 

```
clusterawsadm version: &version.Info{Major:"0", Minor:"6", GitVersion:"v0.6.5", GitCommit:"c7c8ae1ee071afbb33c3b45b0f9adbea149581ce", GitTreeState:"clean", BuildDate:"2021-04-15T16:55:53Z", GoVersion:"go1.13.15", AwsSdkVersion:"v1.36.26", Compiler:"gc", Platform:"linux/amd64"}


clusterctl version: clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.16", GitCommit:"6d7649fc5f407ac48b6d8e3929c97822527c0454", GitTreeState:"clean", BuildDate:"2021-04-16T01:01:02Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

aws --version: aws-cli/2.1.19
```

- Kubernetes version: (use `kubectl version`): v1.20.2
- OS (e.g. from `/etc/os-release`): Ubuntu 20.04

### PR DESCRIPTION
**Pull request by Szymongib**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR contains the following changes:
- Aligns `clusterawsadm bootstrap iam delete-cloudformation-stack` and `clusterawsadm bootstrap iam create-cloudformation-stack` in a way they accept region parameter:
  ```
  config < env < flag
  ```
- Removes call to `ShowStackResources` which always results in error.
- Delete incorrect alias from `delete-cloudformation-stack` command.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2392

**Special notes for your reviewer**:

As suggested in #2392 we can further expand the config resolution to `~/.aws/credentials` for both methods, as it was unintentionally happening right now for `create-cloudformation-stack`.

**Checklist**:

- [x] squashed commits

**Release note**:

```release-note
Align region resolution in create/delete cloudformation stack commands
```

### CODE REVIEW

File: cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go [hunk 1]
```diff
@@ -156,7 +160,7 @@ func deleteCloudFormationStackCmd() *cobra.Command {
 				return err
 			}
 
-			return cfnSvc.ShowStackResources(t.Spec.StackName)
+			return nil
```
Comment (sedefsavas) [inline cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go:163]: We can get a signal from using `WaitUntilStackDeleteComplete ` here, similar to this:
```
klog.V(2).Infof("waiting for stack %q to delete", stackName)
	if err := s.CFN.WaitUntilStackDeleteComplete(&cfn.DescribeStacksInput{StackName: aws.String(stackName)}); err != nil {
		return errors.Wrap(err, "failed to delete AWS CloudFormation stack")
	}
```

File: cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go [hunk 2]
```diff
@@ -92,9 +92,14 @@ func createCloudFormationStackCmd() *cobra.Command {
 				return err
 			}
 
+			if err := resolveTemplateRegion(t, cmd); err != nil {
```
Comment (sedefsavas) [inline cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go:95]: If we remove the err check here, `t.Spec.Region` will be set to empty when there is no env var set, and passing an empty region to `session.Options` makes it resolve region from the default profile.

Same should be done for delete stack logic too.

File: cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go (hunk 1 above)
Comment (Szymongib) [reply cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go:163]: This is already called as part of `DeleteStack` here https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go#L153 https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/5a1266bcf509a4827c1d6d452cf10936bedc44be/cmd/clusterawsadm/cloudformation/service/service.go#L128 so it seems kind of redundant but I can add it if you feel strong about it.

Comment (sedefsavas) [reply cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go:163]: You are right, it is already called.

File: cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go [hunk 3]
```diff
@@ -93,7 +93,7 @@ func createCloudFormationStackCmd() *cobra.Command {
 			}
 
 			if err := resolveTemplateRegion(t, cmd); err != nil {
-				return err
+				fmt.Printf("AWS_REGION env not set and --region flag not provided, default configuration will be used\n")
```
Comment (sedefsavas) [inline cmd/clusterawsadm/cmd/bootstrap/iam/cloudformation.go:96]: nit.
fmt.Println("AWS_REGION env not set and --region flag not provided, default configuration will be used")

**Comment by sedefsavas** [comment]
After squashing commits and addressing minor comment above, this is good to go.

**Pull request merged**